homeassistant/components/sensor/google_wifi.py

```diff
@@ -0,0 +1,170 @@
+    host = config.get(CONF_HOST)
+
+    api = GoogleWifiAPI(host)
+
+    sensors = [GoogleWifiSensor(hass, api, name, condition)
+               for condition in config[CONF_MONITORED_CONDITIONS]]
+
+    add_devices(sensors, True)
+
+
+class GoogleWifiSensor(Entity):
+    """Representation of a Google Wifi sensor."""
+
+    def __init__(self, hass, api, name, variable):
+        """Initialize a Pi-Hole sensor."""
+        self._hass = hass
+        self._api = api
+        self._name = name
+
+        variable_info = MONITORED_CONDITIONS[variable]
+        self._var_name = variable
+        self._var_units = variable_info[1]
+        self._var_icon = variable_info[2]
+        self._var_key = variable_info[3]
+        self._uptime_keys = MONITORED_CONDITIONS[ATTR_UPTIME][3]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "{}_{}".format(self._name, self._var_name)
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._var_icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._var_units
+
+    @property
+    def availiable(self):
+        """Return availiability of goole wifi api."""
+        return self._api.availiable
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if self._var_name == ATTR_LAST_RESTART and self.availiable:
+            uptime = self._api.data[self._uptime_keys[0]][self._uptime_keys[1]]
+            last_restart = dt.now() - timedelta(seconds=uptime)
+            return last_restart.strftime("%Y-%m-%d %H:%M:%S")
+
+        if self.availiable:
+            if (not self._api.data['wan']['ipAddress'] and
+                    self._var_name == ATTR_LOCAL_IP):
+                state_data = STATE_UNKNOWN
+            else:
+                state_data = self._api.data[self._var_key[0]][self._var_key[1]]
+
+            if self._var_name == ATTR_UPTIME:
+                return round(state_data / (24 * 3600), 2)
+            elif self._var_name == ATTR_STATUS:
+                if state_data:
+                    return "Online"
+                return "Offline"
+            elif self._var_name == ATTR_NEW_VERSION:
+                if state_data == '0.0.0.0':
+                    return "Latest"
+            return state_data
+
+        return STATE_UNKNOWN
+
+    def update(self):
+        """Get the latest data from the Google Wifi API."""
+        self._api.update()
+
+
+class GoogleWifiAPI(object):
+    """Get the latest data and update the states."""
+
+    def __init__(self, host):
+        """Initialize the data object."""
+        uri = 'http://'
+        resource = "{}{}{}".format(uri, host, ENDPOINT)
+
+        self._request = requests.Request('GET', resource).prepare()
+        self.data = None
+        self.availiable = True
+        self.update()
+
+    def update(self):
```

Comment:
Use Throttle (https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/util/__init__.py#L233) or you will hit your hup